### PR TITLE
Add support for api version 2022040707071 #72037

### PR DESCRIPTION
**Pull request by robotastic-io**
Add support for api version 2022040707071
#72037

### CODE REVIEW

**Comment by robotastic-io** [comment]
[#72037](https://github.com/home-assistant/core/issues/72037)

File: connect_box/__init__.py [hunk 1]
```diff
@@ -513,8 +526,17 @@ async def _async_initialize_valid_token(self) -> None:
                 headers=self.headers,
                 timeout=10,
             ) as response:
-                await response.text()
+                html = await response.text()
+                apiversion = re.findall("CommonAPI\.js\?v=(\d*)", html)
                 self.token = response.cookies["sessionToken"].value
+                if apiversion and len(apiversion) > 0:
```
Comment (fabaff) [inline connect_box/__init__.py:532]: I guess that this is sufficient to check if the list is not empty.

```suggestion
                if apiversion:
```

File: connect_box/__init__.py [hunk 2]
```diff
@@ -513,8 +526,17 @@ async def _async_initialize_valid_token(self) -> None:
...
+                    self.apiversion = int(apiversion[0])
+                if self.apiversion and self.apiversion >= 20220407070717:
+                    self.cookies["sessionToken"] = f"{self.token}"
+                else:
+                    self.cookies = None
+                    self.request_token_string = f"token=${self.token}&"
```
Comment (fabaff) [inline connect_box/__init__.py:538]: Prefixing the value for the token with a `$` will most likely lead to login failures an router which the default firmware.

File: connect_box/__init__.py [hunk 3]
```diff
@@ -525,18 +547,22 @@ async def _async_do_login_with_password(self, function: int) -> None:
...
             ) as response:
-                await response.text()
+                html = await response.text()
 
                 if response.status != 200:
                     _LOGGER.warning("Login error with code %d", response.status)
                     self.token = None
                     raise exceptions.ConnectBoxLoginError()
                 self.token = response.cookies["sessionToken"].value
+                if self.apiversion >= 20220407070717:
```
Comment (fabaff) [inline connect_box/__init__.py:563]: This could end in a comparison of `None` with an integer.

File: connect_box/__init__.py (hunk 1 above)
Comment (robotastic-io) [reply connect_box/__init__.py]: Changed as suggested

**Comment by robotastic-io** [comment]
I just noticed a possible issue for older versions. Will get back with a fix

**Comment by robotastic-io** [comment]
@fabaff Please review

**Comment by tams9** [comment]
Hi, Can you help me, how to apply the changes and try the new code?
First I thought that I just need to modify a file, but I cannot find any connect box related folder or file in Home Assistant server.

**Comment by fabaff** [comment]
> Hi, Can you help me, how to apply the changes and try the new code?

Clone the repo, checkout this branch and run [`example.py`](https://github.com/home-assistant-ecosystem/python-connect-box/blob/master/example.py).

> First I thought that I just need to modify a file, but I cannot find any connect box related folder or file in Home Assistant server.

To test it with Home Assistant you would need to create a `custom_component` which is using the changes from this branch.

**Comment by robotastic-io** [comment]
As Vodafone recently changed my router I can no longer test/support this. It was working a couple of month ago. As fabaff mentioned you can use the code to create your own custom component. Maybe someone is willing to pick this up.

**Pull request closed without merging**